esphome/components/climate/climate.cpp

```diff
@@ -494,5 +494,152 @@ void ClimateDeviceRestoreState::apply(Climate *climate) {
+  return set_alternative(this->custom_preset, this->preset, preset);
+}
+
+static const char *mode_to_string(ClimateMode mode) {
+  switch (mode) {
+    case ClimateMode::CLIMATE_MODE_AUTO:
+      return "AUTO";
+    case ClimateMode::CLIMATE_MODE_COOL:
+      return "COOL";
+    case ClimateMode::CLIMATE_MODE_DRY:
+      return "DRY";
+    case ClimateMode::CLIMATE_MODE_FAN_ONLY:
+      return "FAN_ONLY";
+    case ClimateMode::CLIMATE_MODE_HEAT:
+      return "HEAT";
+    case ClimateMode::CLIMATE_MODE_HEAT_COOL:
+      return "HEAT_COOL";
+    case ClimateMode::CLIMATE_MODE_OFF:
+      return "OFF";
+  }
+  return "UNKNOWN";
+}
+
+static const char *fanmode_to_string(ClimateFanMode mode) {
+  switch (mode) {
+    case ClimateFanMode::CLIMATE_FAN_AUTO:
+      return "AUTO";
+    case ClimateFanMode::CLIMATE_FAN_DIFFUSE:
+      return "DIFFUSE";
+    case ClimateFanMode::CLIMATE_FAN_FOCUS:
+      return "FOCUS";
+    case ClimateFanMode::CLIMATE_FAN_HIGH:
+      return "HIGH";
+    case ClimateFanMode::CLIMATE_FAN_LOW:
+      return "LOW";
+    case ClimateFanMode::CLIMATE_FAN_MEDIUM:
+      return "MEDIUM";
+    case ClimateFanMode::CLIMATE_FAN_MIDDLE:
+      return "MIDDLE";
+    case ClimateFanMode::CLIMATE_FAN_OFF:
+      return "OFF";
+    case ClimateFanMode::CLIMATE_FAN_ON:
+      return "ON";
+  }
+  return "UNKNOWN";
+}
+
+static const char *swingmode_to_string(ClimateSwingMode mode) {
+  switch (mode) {
+    case ClimateSwingMode::CLIMATE_SWING_BOTH:
+      return "BOTH";
+    case ClimateSwingMode::CLIMATE_SWING_HORIZONTAL:
+      return "HORIZONTAL";
+    case ClimateSwingMode::CLIMATE_SWING_OFF:
+      return "OFF";
+    case ClimateSwingMode::CLIMATE_SWING_VERTICAL:
+      return "VERTICAL";
+  }
+  return "UNKNOWN";
+}
+
+static const char *preset_to_string(ClimatePreset preset) {
+  switch (preset) {
+    case ClimatePreset::CLIMATE_PRESET_ACTIVITY:
+      return "ACTIVITY";
+    case ClimatePreset::CLIMATE_PRESET_AWAY:
+      return "AWAY";
+    case ClimatePreset::CLIMATE_PRESET_BOOST:
+      return "BOOST";
+    case ClimatePreset::CLIMATE_PRESET_COMFORT:
+      return "COMFORT";
+    case ClimatePreset::CLIMATE_PRESET_ECO:
+      return "ECO";
+    case ClimatePreset::CLIMATE_PRESET_HOME:
+      return "HOME";
+    case ClimatePreset::CLIMATE_PRESET_NONE:
+      return "NONE";
+    case ClimatePreset::CLIMATE_PRESET_SLEEP:
+      return "SLEEP";
+  }
+  return "UNKNOWN";
+}
+
+void Climate::dump_traits(const char *tag) {
+  auto traits = this->get_traits();
+  ESP_LOGCONFIG(tag, "ClimateTraits:");
+  ESP_LOGCONFIG(tag, "  [x] Visual settings:");
+  ESP_LOGCONFIG(tag, "      - Min: %.1f", traits.get_visual_min_temperature());
+  ESP_LOGCONFIG(tag, "      - Max: %.1f", traits.get_visual_max_temperature());
+  ESP_LOGCONFIG(tag, "      - Step: %.1f", traits.get_visual_temperature_step());
+  if (traits.get_supports_current_temperature())
+    ESP_LOGCONFIG(tag, "  [x] Current temperature");
```

Comment:
```suggestion
    ESP_LOGCONFIG(tag, "  [x] Supports Current temperature");
```

Otherwise probably hard to understand what that's supposed to mean